python/pyspark/ml/feature.py

```diff
@@ -408,35 +408,86 @@ class CountVectorizer(JavaEstimator, HasInputCol, HasOutputCol, JavaMLReadable,
     True
-    >>> loadedCv.getMinTF() == cv.getMinTF()
+    >>> loadedCv.getMinTF() == cv1.getMinTF()
     True
-    >>> loadedCv.getVocabSize() == cv.getVocabSize()
+    >>> loadedCv.getVocabSize() == cv1.getVocabSize()
     True
     >>> modelPath = temp_path + "/count-vectorizer-model"
-    >>> model.save(modelPath)
+    >>> model1.save(modelPath)
     >>> loadedModel = CountVectorizerModel.load(modelPath)
-    >>> loadedModel.vocabulary == model.vocabulary
+    >>> loadedModel.vocabulary == model1.vocabulary
     True
+    >>> df2 = spark.createDataFrame(
+    ...    [(0, ["a", "b", "c", "d"]), (1, ["a", "b", "c",]),(2, ["a", "b"]),(3, ["a"]),],
+    ...    ["label", "raw"])
+    >>> cv2 = CountVectorizer(inputCol="raw", outputCol="vectors", maxDF=3)
+    >>> model2 = cv2.fit(df2)
+    >>> model2.transform(df2).show(truncate=False)
+    +-----+------------+-------------------------+
+    |label|raw         |vectors                  |
+    +-----+------------+-------------------------+
+    |0    |[a, b, c, d]|(3,[0,1,2],[1.0,1.0,1.0])|
+    |1    |[a, b, c]   |(3,[0,1],[1.0,1.0])      |
+    |2    |[a, b]      |(3,[0],[1.0])            |
+    |3    |[a]         |(3,[],[])                |
+    +-----+------------+-------------------------+
+    ...
+    >>> cv3 = CountVectorizer(inputCol="raw", outputCol="vectors", maxDF=0.75)
+    >>> model3 = cv3.fit(df2)
+    >>> model3.transform(df2).show(truncate=False)
+    +-----+------------+-------------------------+
+    |label|raw         |vectors                  |
+    +-----+------------+-------------------------+
+    |0    |[a, b, c, d]|(3,[0,1,2],[1.0,1.0,1.0])|
+    |1    |[a, b, c]   |(3,[0,1],[1.0,1.0])      |
+    |2    |[a, b]      |(3,[0],[1.0])            |
+    |3    |[a]         |(3,[],[])                |
+    +-----+------------+-------------------------+
+    ...
+    >>> cv4 = CountVectorizer(inputCol="raw", outputCol="vectors", minDF=2, maxDF=3)
+    >>> model4 = cv4.fit(df2)
+    >>> model4.transform(df2).show(truncate=False)
+    +-----+------------+-------------------+
+    |label|raw         |vectors            |
+    +-----+------------+-------------------+
+    |0    |[a, b, c, d]|(2,[0,1],[1.0,1.0])|
+    |1    |[a, b, c]   |(2,[0,1],[1.0,1.0])|
+    |2    |[a, b]      |(2,[0],[1.0])      |
+    |3    |[a]         |(2,[],[])          |
+    +-----+------------+-------------------+
+    ...
+    >>> cv5 = CountVectorizer(inputCol="raw", outputCol="vectors", minDF=0.5, maxDF=0.75)
+    >>> model5 = cv5.fit(df2)
+    >>> model5.transform(df2).show(truncate=False)
+    +-----+------------+-------------------+
+    |label|raw         |vectors            |
+    +-----+------------+-------------------+
+    |0    |[a, b, c, d]|(2,[0,1],[1.0,1.0])|
+    |1    |[a, b, c]   |(2,[0,1],[1.0,1.0])|
+    |2    |[a, b]      |(2,[0],[1.0])      |
+    |3    |[a]         |(2,[],[])          |
+    +-----+------------+-------------------+
+    ...
```

Comment:
I think this is too much to put as a doctest.  Instead, can you just add a unit test in ml/tests.py?  I think you just need 2 transforms, one with an integer value of `maxDF` > 1 and one as a fractional value.  Also, I don't think your test data actually uses the `maxDF` filtering.